includes/ucf-post-list-common.php

```diff
@@ -230,6 +230,41 @@ public static function get_image_or_fallback( $item, $size='large' ) {
 			return $img;
 		}
 
+		/**
+		 * Retrieves the post excerpt and sets a character limit
+		 *
+		 * @author Mike Setzer
+		 * @since 2.0.8
+		 * @param object $item | object containing the WP Post
+		 * @param int $max_words | Maximum characters displayed from excerpt
+		 * @return string $excerpt | Excerpt text
+		 **/
+		public static function get_excerpt ( $item, $max_words) {
+			$excerpt = null;
+
+			//If excerpt is left empty, the post_content will be taken instead
+			if ( empty( $item->post_excerpt ) ) {
+				//This scrubs HTML tags and WordPress block tags from the content, before calculating strlen
+				$filtered_content = wp_strip_all_tags($item->post_content);
+			} else {
+				$filtered_content = wp_strip_all_tags($item->post_excerpt);
+			}
+
+			$filtered_words = explode (' ', $filtered_content, $max_words+1);
+			if ($max_words > 0) {
+				if (count($filtered_words) >= $max_words ) {
+					array_pop($filtered_words);
+					$excerpt = implode(" ", $filtered_words) . '...';
+				} else {
+					$excerpt = implode(" ", $filtered_words);
+				}
+			} else {
+				$excerpt = implode (" ", $filtered_words);
+			}
+
+			return $excerpt;
+		}
+
```

Comment:
You might be able to simplify this down using [`wp_trim_words`](https://developer.wordpress.org/reference/functions/wp_trim_words/).

Reply:
Tested wp_trim_words but it would only filter html tags and leave shortcodes. Tried wp_trim_words on strings that had already been run through the preg_replace function used to strip shortcodes below.

$stripped_content = preg_replace("~(?:\[/?)[^/\]]+/?\]~s", '', $filtered_content);